spec/support/shared_linter_context.rb

```diff
@@ -10,9 +10,9 @@
 
   let(:config) { options[:config].for_linter(described_class) }
 
+  let(:document) { HamlLint::Document.new(normalize_indent(haml), options) }
```

Comment:
This change lets us inject the document for the test, letting us set the file name for a test.